src/AzureAppConfigurationImpl.ts

```diff
@@ -479,17 +337,141 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
     }
 
     /**
-     * Refresh key-values.
+     * Loads configuration settings from App Configuration, either key-value settings or feature flag settings.
+     * Additionally, updates the `pageEtags` property of the corresponding @see PagedSettingSelector after loading.
+     *
+     * @param loadFeatureFlag - Determines which type of configurationsettings to load:
+     *                          If true, loads feature flag using the feature flag selectors;
+     *                          If false, loads key-value using the key-value selectors. Defaults to false.
+     */
+    async #loadConfigurationSettings(loadFeatureFlag: boolean = false): Promise<ConfigurationSetting[]> {
+        const selectors = loadFeatureFlag ? this.#ffSelectors : this.#kvSelectors;
+        const funcToExecute = async (client) => {
+            const loadedSettings: ConfigurationSetting[] = [];
+            // deep copy selectors to avoid modification if current client fails
+            const selectorsToUpdate = JSON.parse(
+                JSON.stringify(selectors)
+            );
+
+            for (const selector of selectorsToUpdate) {
+                const listOptions: ListConfigurationSettingsOptions = {
+                    keyFilter: selector.keyFilter,
+                    labelFilter: selector.labelFilter
+                };
+
+                const pageEtags: string[] = [];
+                const pageIterator = listConfigurationSettingsWithTrace(
+                    this.#requestTraceOptions,
+                    client,
+                    listOptions
+                ).byPage();
+                for await (const page of pageIterator) {
+                    pageEtags.push(page.etag ?? "");

```

Comment:
What happens when the page etag is null and we just set it to empty string here?

Reply:
AFAIK, page.etag shall not be null, even empty page should have a etag. But pageEtags's type is `string[]`. 
This is Yan's original implementation, I didn't change it: https://github.com/Azure/AppConfiguration-JavaScriptProvider/blob/main/src/AzureAppConfigurationImpl.ts#L275 @Eskibear

Reply:
I don't remember clearly why but probably it's for simplicity, then we don't have to do null check later when comparing the etags. Alternatively, if `page.etag` can be null (from sdk declaration and implementation), we can also declare `pageEtags` as `(string | null)[]` or `Array<string | null>`.
After all, it depends on how we want to distinguish whether etag is null or empty. Is empty string a valid etag?

Reply:
I don't think there will be anycase when etag is empty or null. Empty page should have a etag `00wWELzclWef22f82qcUDI7FbJweVOoKWZDA_JoUeD0`

Reply:
Service will not respond with null or empty Etag (unless using older API versions).

Reply:
Thanks for clarifying!

Reply:
I will keep the current implementation.